near/nearcore
# Store restored receipts in the storage

**Idea**
~~Return `restored_receipts` from `apply_transactions` to indicate that some receipts were added in runtime due to migration, so they will not be lost in chain and could be accessed by Indexer.~~

Create new database migration to add receipts missing in store directly into it.

**Testing**
~~Extend existing `protocol_feature_restore_receipts_after_fix_tests` by running `get_receipt` method.~~

- setup archival mainnet node
- run `./target/release/neard --home ~/.near init`, observe migration:
```
$ ./target/debug/neard --home ~/.near run
Jul 09 12:57:59.635  INFO near: Version: 1.19.1, Build: c8e49bb5, Latest Protocol: 45
Jul 09 12:57:59.760  INFO near: Opening store database at "/home/Aleksandr1/.near/data"
apply_store_migrations
Jul 09 12:57:59.781  INFO near: Migrate DB from version 22 to 23
```
- stop node, access store using separate script and ensure that receipts are saved in it
```
for receipt in receipts {
    eprintln!("{}", receipt.get_hash());
    chain_store.get_receipt(&receipt.get_hash()).unwrap().unwrap();
}
```
